src/main/java/com/notably/model/BlockModel.java

```diff
@@ -0,0 +1,53 @@
+package com.notably.model;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.model.block.Block;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeItem;
+
+/**
+ * API of the BlockModel component.
+ */
+public interface BlockModel {
+    /**
+     * Gets the BlockTree currently in use by Notably.
+     */
+    BlockTree getBlockTree();
+
+    /**
+     * Check if currently open block has the specified child block.
+     */
+    boolean hasChild(Block b);
```

Comment:
Is this method needed? I couldn't think of a use case for it

Reply:
Perhaps a better method would be `hasPath(AbsolutePath path)`? But still, I couldn't think of a use case for it too

Reply:
This method would be useful to check for blocks with the same title under the same directory. 
As we don't allow such to happen, an exception would be thrown if the user decides to do that.

Reply:
I see, should we reframe this as `hasPath(AbsolutePath path)` then? In a way, `hasChild` implicitly mean that the child has to belong to the `currentlyOpenBlock` (it requires the state for the currently active block), which might not be very intuitive.

`hasPath`, on the other hand, works without any relation to that context. @ljiazh3ng @firzanarmani what do you guys think?

Reply:
I also feel `hasPath` is more intuitive! But yeet what do you guys think? @ljiazh3ng @firzanarmani

Reply:
Sounds good lets change it to `hasPath` instead @firzanarmani.